services/ui-src/src/components/reports/ModalDrawerReportPage.tsx

```diff
@@ -33,6 +39,18 @@ export const ModalDrawerReportPage = ({ route }: Props) => {
   const { report, updateReport } = useContext(ReportContext);
   const reportFieldDataEntities = report?.fieldData[entityType] || [];
 
+  // create drawerForm from json with repeated fields
+  const drawerForm = { ...drawerFormJson };
+  const formContainsFieldsToRepeat = drawerFormJson.fields.find(
+    (field: FormField) => field.repeat
+  );
+  if (formContainsFieldsToRepeat) {
+    drawerForm.fields = createRepeatedFields(
+      drawerFormJson.fields,
+      report?.fieldData
+    );
+  }
```

Comment:
Note: Someday I think we'll want to move this elsewhere (a piece of FormProvider Context state? 🤔 ) but that will require some additional refactoring that is for sure out of scope right now. 👍